File: iosSDK/text/Refresh/Refresh/ViewController.m
//
//  ViewController.m
//  Refresh
//
//  Created by  wj on 16/5/4.
//  Copyright © 2016年 gzq. All rights reserved.
//

#import "ViewController.h"
#import "MyTableViewCell.h"
#import "AFNetworking.h"
#import "MJRefresh.h"

#define IMAHEURL @"http://iappfree.candou.com:8080/free/applications/limited?currency=rmb&page=%ld"

@interface ViewController (){
    
    UITableView *_tableView;
    
    NSMutableArray *_dataArray;
    
    NSInteger _page;
    
    BOOL _isPulling;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _isPulling = NO;
    
    _dataArray = [[NSMutableArray alloc]init];
    
    _page = 1;
    
    [self creatTableView];
    
    [self creatData];
    
    [self creatRefresh];
    
    
    // Do any additional setup after loading the view, typically from a nib.
}
-(void)creatTableView{
    
    _tableView = [[UITableView alloc]initWithFrame:self.view.bounds style:UITableViewStylePlain];
    
    _tableView.delegate = self;
    
    _tableView.dataSource = self;
    
    [self.view addSubview:_tableView];
}
-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
     NSLog(@"%lu",(unsigned long)_dataArray.count);
    return _dataArray.count;
    
}
-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    MyTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ID"];
    if (!cell) {
        cell = [[MyTableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"ID"];
    }
    MyModel *model = _dataArray[indexPath.row];
    NSLog(@"%@",model.name);
    [cell configCellWithMode:model];
    return cell;
}
-(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return  90;
}
-(void)creatData{
    
    AFHTTPRequestOperationManager *manager = [AFHTTPRequestOperationManager manager];
    
    manager.responseSerializer = [AFHTTPResponseSerializer serializer];
    
    [manager GET:[NSString stringWithFormat:IMAHEURL,_page] parameters:nil success:^(AFHTTPRequestOperation *operation, id responseObject) {
        NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:responseObject options:NSJSONReadingAllowFragments error:nil];
        NSArray *array = [dic objectForKey:@"applications"];
        
        if (_isPulling == YES) {
            [_dataArray removeAllObjects];
        }
        
        for(NSDictionary *dic in array){
            MyModel *model = [[MyModel alloc]init];
            
//            model.iconUrl = [dic objectForKey:@"iconUrl"];
//            model.name = [dic objectForKey:@"name"];
//            model.applicationId = [dic objectForKey:@"applicationId"];
            
            [model setValuesForKeysWithDictionary:dic];
            
            [_dataArray addObject:model];
            
             
        }
        
        if (_isPulling ==YES) {
            [_tableView headerEndRefreshing];
        }else{
            [_tableView footerEndRefreshing];
        }
         [_tableView reloadData];
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        NSLog(@"%@",error);
    }];
    
}
-(void)creatRefresh{
    
    [_tableView addHeaderWithTarget:self action:@selector(pullRrefresh)];
    
    [_tableView addFooterWithTarget:self action:@selector(pushRefresh)];
}

-(void)pullRrefresh{
    
    _isPulling = YES;
    
    _page = 1;
    
    [self creatData];
}
-(void)pushRefresh{
    
    _isPulling = NO;
    
    _page ++;
    
    [self creatData];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iosSDK/TouchExplorer/TouchExplorer/ViewController.swift
//
//  ViewController.swift
//  TouchExplorer
//
//  Created by  wj on 16/1/29.
//  Copyright © 2016年 gzq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var messageLabel: UILabel!
    @IBOutlet weak var tapsLabel: UILabel!
    @IBOutlet weak var touchesLabel: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func updateLabelsFromTouches(touches:Set<UITouch>){
        let touch = (touches as NSSet).anyObject() as! UITouch
        let numTaps = touch.tapCount
        let tapsMessage = "\(numTaps)次轻点"
        tapsLabel.text = tapsMessage
        
        let numTouches = touches.count
        let touchMsg = "\(numTouches)根手指"
        touchesLabel.text = touchMsg
    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        messageLabel.text = "开始接触"
        updateLabelsFromTouches((event?.allTouches())!)
    }
    override func touchesCancelled(touches: Set<UITouch>?, withEvent event: UIEvent?) {
        messageLabel.text = "触摸取消"
        updateLabelsFromTouches((event?.allTouches())!)
    }
    override func touchesEnded(touches: Set<UITouch>, withEvent event: UIEvent?) {
        messageLabel.text = "接触结束"
        updateLabelsFromTouches((event?.allTouches())!)
    }
    override func touchesMoved(touches: Set<UITouch>, withEvent event: UIEvent?) {
        messageLabel.text = "手势"
        updateLabelsFromTouches((event?.allTouches())!)
    }
}

